scripts/test.sh

```diff
@@ -5,10 +5,11 @@ set -ex
 export PATH="$PATH:/snap/bin"
 
 # Grab the action parameter values
-FOCUS=$(action-get focus)
-SKIP=$(action-get skip)
-PARALLELISM=$(action-get parallelism)
-EXTRA_ARGS=$(action-get extra)
+FOCUS=${1}
+SKIP=${2}
+PARALLELISM=${3}
+TIMEOUT=${4}
+EXTRA_ARGS=${5}
```

Comment:
you may also need to capture those `action-fail` and `action-set` calls in your `charm.py` calls? I'm curious to see if the juju agent cares if they come from subprocess or not. 

my fear is the the subprocess will finish, write to action-fail and stuff... then return from the subprocess -- and your charm code will clean up the action differently